module/VuFindSearch/src/VuFindSearch/Backend/Solr/Connector.php

```diff
@@ -494,6 +503,15 @@ protected function send(HttpClient $client)
         );
 
         if (!$response->isSuccess()) {
+            // Return a more detailed error message for a 400 error:
+            if ($response->getStatusCode() === 400) {
+                $json = json_decode($response->getBody(), true);
+                throw new RequestErrorException(
+                    '400 ' . $response->getReasonPhrase() . ' ' . $json['error']['msg'],
```

Comment:
Should we account for the possibility that `$json['error']['msg']` is undefined?

Reply:
Ah, yes, of course. Check added.